src/java.base/share/classes/java/time/Instant.java

```diff
@@ -1169,15 +1169,25 @@ private long nanosUntil(Instant end) {
     }
 
     private long microsUntil(Instant end) {
-        long secsDiff = Math.subtractExact(end.seconds, seconds);
-        long totalMicros = Math.multiplyExact(secsDiff, MICROS_PER_SECOND);
-        return Math.addExact(totalMicros, (end.nanos - nanos) / 1000);
+        long microsDiff = Math.multiplyExact(end.seconds - seconds, MICROS_PER_SECOND);
+        long nanosDiff = end.nanos - nanos;
```

Comment:
FWIW, `nanosDiff` could be declared `int` to slightly speed up the division later in the code, while still not risking overflows.

Similarly in `millisUntil()`.